File: apps/zxtune-android/src/app/zxtune/playback/Iterator.java
/**
 *
 * @file
 *
 * @brief Playback iterator interface
 *
 * @author vitamin.caig@gmail.com
 *
 */

package app.zxtune.playback;

import app.zxtune.Releaseable;

public interface Iterator extends Releaseable {

  /**
   *  Retrieve current position' item
   *  @return Newly loaded item
   */
  PlayableItem getItem();

  /**
   *  Move to next position
   *  @return true if successfully moved, else getItem will return the same item
   */
  boolean next();
  
  /**
   *  Move to previous position
   *  @return true if successfully moved, else getItem will return the same item
   */
  boolean prev();
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Iterator.kt
/**
 *
 * @file
 *
 * @brief Playback iterator interface
 *
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.playback

interface Iterator {
    /**
     * Retrieve current position' item
     * @return Newly loaded item
     */
    val item: PlayableItem

    /**
     * Move to next position
     * @return true if successfully moved, else getItem will return the same item
     */
    fun next(): Boolean

    /**
     * Move to previous position
     * @return true if successfully moved, else getItem will return the same item
     */
    fun prev(): Boolean
}
